client/packages/inventory/src/Stocktake/context/stocktakeLineError.tsx

```diff
@@ -0,0 +1,63 @@
+import { PropsWithChildrenOnly, RecordWithId } from '@common/types';
+import React, { createContext, useContext, useState } from 'react';
+import {
+  AdjustmentReasonNotProvidedErrorFragment,
+  AdjustmentReasonNotValidErrorFragment,
+  SnapshotCountCurrentCountMismatchErrorFragment,
+  StockLineReducedBelowZeroErrorFragment,
+} from '../api';
+
+export type StocktakeLineError =
+  | AdjustmentReasonNotProvidedErrorFragment
+  | StockLineReducedBelowZeroErrorFragment
+  | AdjustmentReasonNotValidErrorFragment
+  | SnapshotCountCurrentCountMismatchErrorFragment;
+
+const useStocktakeLineErrors = () => {
+  const [errors, setErrors] = useState<{
+    [id: string]: StocktakeLineError | undefined;
+  }>({});
+
+  const getError = ({ id }: RecordWithId): StocktakeLineError | undefined => {
+    return errors[id];
+  };
+
+  const setError = (id: string, error: StocktakeLineError) => {
+    setErrors(errors => ({ ...errors, [id]: error }));
+  };
+
+  const unsetError = (id: string) => {
+    setErrors(errors => ({ ...errors, [id]: undefined }));
+  };
+
+  const unsetAll = () => {
+    setErrors({});
+  };
+
+  return { errors, setError, setErrors, getError, unsetError, unsetAll };
+};
+export type UseStocktakeLineErrors = ReturnType<typeof useStocktakeLineErrors>;
```

Comment:
rather then defining the type of each individual method, can use return type of the main state hook